test/utils/structs/EnumerableMap.test.js

```diff
@@ -14,132 +15,86 @@ const getMethods = (mock, fnSigs) => {
-  });
-
-  // Bytes32ToBytes32Map
-  describe('Bytes32ToBytes32Map', function () {
-    const fixture = async () => {
-      const mock = await ethers.deployContract('$EnumerableMap');
+const testTypes = [formatType('bytes32', 'bytes32'), ...TYPES];
 
-      const [keyA, keyB, keyC] = randomArray(generators.bytes32);
-      const [valueA, valueB, valueC] = randomArray(generators.bytes32);
-
-      const methods = getMethods(mock, {
-        set: '$set(uint256,bytes32,bytes32)',
-        get: '$get_EnumerableMap_Bytes32ToBytes32Map(uint256,bytes32)',
-        tryGet: '$tryGet_EnumerableMap_Bytes32ToBytes32Map(uint256,bytes32)',
-        remove: '$remove_EnumerableMap_Bytes32ToBytes32Map(uint256,bytes32)',
-        length: '$length_EnumerableMap_Bytes32ToBytes32Map(uint256)',
-        at: '$at_EnumerableMap_Bytes32ToBytes32Map(uint256,uint256)',
-        contains: '$contains_EnumerableMap_Bytes32ToBytes32Map(uint256,bytes32)',
-        keys: '$keys_EnumerableMap_Bytes32ToBytes32Map(uint256)',
-      });
+async function fixture() {
+  const mock = await ethers.deployContract('$EnumerableMap');
 
-      return { mock, keyA, keyB, keyC, valueA, valueB, valueC, methods };
-    };
-
-    beforeEach(async function () {
-      Object.assign(this, await loadFixture(fixture));
-    });
-
-    shouldBehaveLikeMap(ethers.ZeroHash, 'bytes32', {
-      setReturn: 'return$set_EnumerableMap_Bytes32ToBytes32Map_bytes32_bytes32',
-      removeReturn: 'return$remove_EnumerableMap_Bytes32ToBytes32Map_bytes32',
-    });
-  });
-
-  // UintToUintMap
-  describe('UintToUintMap', function () {
-    const fixture = async () => {
-      const mock = await ethers.deployContract('$EnumerableMap');
-
-      const [keyA, keyB, keyC] = randomArray(generators.uint256);
-      const [valueA, valueB, valueC] = randomArray(generators.uint256);
-
-      const methods = getMethods(mock, {
-        set: '$set(uint256,uint256,uint256)',
-        get: '$get_EnumerableMap_UintToUintMap(uint256,uint256)',
-        tryGet: '$tryGet_EnumerableMap_UintToUintMap(uint256,uint256)',
-        remove: '$remove_EnumerableMap_UintToUintMap(uint256,uint256)',
-        length: '$length_EnumerableMap_UintToUintMap(uint256)',
-        at: '$at_EnumerableMap_UintToUintMap(uint256,uint256)',
-        contains: '$contains_EnumerableMap_UintToUintMap(uint256,uint256)',
-        keys: '$keys_EnumerableMap_UintToUintMap(uint256)',
-      });
+  const values = mapValues(
+    {
+      uint256: generators.uint256,
+      address: generators.address,
+      bytes32: generators.bytes32,
+    },
+    randomArray,
+  );
 
-      return { mock, keyA, keyB, keyC, valueA, valueB, valueC, methods };
-    };
+  const zeroValue = {
+    uint256: 0n,
+    address: ethers.ZeroAddress,
+    bytes32: ethers.ZeroHash,
+  };
+
+  const env = Object.fromEntries(
+    testTypes.map(({ name, keyType, valueType }) => [
+      name,
+      {
+        keyType,
+        keys: values[keyType],
+        values: values[valueType],
```

Comment:
Maybe we can drop the `values` object. Regenerating isn't that slow, and it make the tests more "independant"
```suggestion
        keys: randomArray(generators[keyType]),
        values: randomArray(generators[valueType]),
```